shared/tab-bar/index.render.desktop.js

```diff
@@ -1,132 +1,110 @@
 // @flow
 
-import React, {Component} from 'react'
-import {Box, TabBar, Avatar} from '../common-adapters'
-import {TabBarButton, TabBarItem} from '../common-adapters/tab-bar'
+import React from 'react'
+import {Box, Avatar} from '../common-adapters'
+import {TabBarButton} from '../common-adapters/tab-bar'
 import {globalStyles, globalColors} from '../styles'
 import flags from '../util/feature-flags'
+import {
+  chatTab,
+  profileTab,
+  peopleTab,
+  folderTab,
+  devicesTab,
+  searchTab,
+  settingsTab,
+} from '../constants/tabs'
 
-import {profileTab, peopleTab, folderTab, devicesTab, settingsTab, chatTab, tabPrettify} from '../constants/tabs'
-
-import type {Tab} from '../constants/tabs'
-import type {IconType} from '../common-adapters/icon'
 import type {Props} from './index.render'
 
-const icons: {[key: Tab]: IconType} = {
-  [peopleTab]: 'iconfont-people',
-  ...(flags.tabChatEnabled ? {[chatTab]: 'iconfont-chat'} : null),
-  [folderTab]: 'iconfont-folder',
-  [devicesTab]: 'iconfont-device',
-  [settingsTab]: 'iconfont-settings',
-}
-
-export type SearchButton = 'TabBar:searchButton'
-export const searchButton = 'TabBar:searchButton'
-
-function tabToIcon (t: Tab): IconType {
-  return icons[t]
-}
-
-export default class TabBarRender extends Component<void, Props, void> {
-  _renderSearch (onClick: () => void) {
-    const source = {type: 'nav', icon: 'iconfont-nav-search'}
-    const button = (
+export default function TabBar ({selectedTab, onTabClick, username, badgeNumbers}: Props) {
+  const avatar = (
+    <Avatar
+      size={32}
+      onClick={() => onTabClick(profileTab)}
+      username={username}
+      borderColor={selectedTab === profileTab ? globalColors.white : globalColors.blue3_40}
+    />
+  )
+
+  return (
+    <Box style={stylesTabBar}>
       <TabBarButton
```

Comment:
This style is less DRY, but it's much clearer exactly what will be displayed onscreen. I considered breaking some of the repeated sections into a helper function, but I felt that would compromise on the clarity.